Classes/common/Encryption/FMDatabase+EncryptionKey.h

```diff
@@ -0,0 +1,47 @@
+//
+//  FMDatabase+EncryptionKey.h
+//  CloudantSync
+//
+//  Created by Enrique de la Torre Fernandez on 27/04/2015.
+//  Copyright (c) 2015 IBM Cloudant. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+//  except in compliance with the License. You may obtain a copy of the License at
+//  http://www.apache.org/licenses/LICENSE-2.0
+//  Unless required by applicable law or agreed to in writing, software distributed under the
+//  License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+//  either express or implied. See the License for the specific language governing permissions
+//  and limitations under the License.
+//
+
+#import "FMDatabase.h"
+
+#import "CDTEncryptionKeyProvider.h"
+
+extern NSString *const FMDatabaseEncryptionKeyErrorDomain;
+
+/**
+ Errors for 'setKeyWithProvider:error:'
+ */
+typedef NS_ENUM(NSInteger, FMDatabaseEncryptionKeyError) {
+    FMDatabaseEncryptionKeyErrorKeyNotSet,
+    FMDatabaseEncryptionKeyErrorNoKeyProvidedForEncryptedDB,
+    FMDatabaseEncryptionKeyErrorWrongKeyOrDBNotEncrypted
+};
```

Comment:
'setKeyWithProvider:error:' is used in 2 different places:
* `CDTQIndexManager`.- It defines its own set of errors: `CDTQQueryError`. I can extend this type and if 'setKeyWithProvider:error:'  fails, parse the error returned to the new errors in `CDTQQueryError`.
* `TD_Database`.- It does not pass the error returned by 'setKeyWithProvider:error:' to other methods in the library, to say in another way, it is self-contained so there is no need to declare the errors in `FMDatabase+EncryptionKey` in other place.